### ISSUE TITLE
[FEATURE] Need status information tracking and complete notification for zero downtime upgrade 

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#1852]
**Describe the bug**

The zero downtime upgrade page need to provide more information for user tracking each component upgrade status.
And notify user after upgrade complete, without this feature user have no idea whether the upgrade is finished or not.

![image](https://user-images.githubusercontent.com/29251855/150490279-14abe188-923f-4ea1-97e0-aaf85ab37cbb.png)
 
**To Reproduce**
Do zero downtime upgrade from v1.0 to latest master head, steps refer to #1022 
1. Prepare 3 nodes harvester with 1.0.0 GA
2. Prepare latest master head iso 
3. Upload iso file to http or file server
4. ssh to harvester controller node 
5. Create the versions.yaml 
    ```
    cat > versions.yaml <<EOF
    apiVersion: harvesterhci.io/v1beta1
    kind: Version
    metadata:
      name: 1.0.2
      namespace: harvester-system
    spec:
      isoChecksum: 2765d89e5e8024347c81d36cc84b7168bf358a3a2f4da0c63164ad834a94bcbbaa13af4900aad4caa0a27550e4b4db5bf2c4726daa88664378e0c581a79ad2cb
      isoURL: http://10.10.0.1/harvester/harvester.iso
      minUpgradableVersion: 0.3.0
      releaseDate: "20211231"
      tags:
      - dev
      - test
    EOF
    ```
6. Run kubectl create -f versions.yaml
7. Refresh harvester page
8. Click the Upgrade button
9. Click the green circle to display upgrade status page 
10. Check the status and wait until zero downtime upgrade finish

**Expected behavior**

Upgrade dashboard should provide upgrade progress or error message of the following  
1. os status
2. rke2 status
3. rancher status
4. harvester and monitoring status
5. Shows the overall status of the upgrade 

Also keep the upgrade dashboard page display after upgrade complete with notification like `Upgrade Complete`

**Support bundle**

**Environment:**
 - Harvester ISO version: v1.0.0 to latest master head  
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 3 nodes harvester on provo bare machines

**Additional context**

**Comment by bk201** [comment]
We also need the ability to access jobs logs during an upgrade. The job logs can be accessed from Kubernetes, but they might be deleted at a later time automatically.

**Comment by TachunLin** [comment]
Verified on `master-fd872c99-head` (03/09). Found a truncation on upgrade dashboard and an enhancement suggestion. 
Move it to implement pipeline. 

@bk201, @WuJun2016, I suggest we can improve them to provide better user experience WDUT. 

### Result
1. The rancher version value truncated and exceed the panel  
  ![image](https://user-images.githubusercontent.com/29251855/157442425-852df417-2b58-4fc4-9522-6228a8d80571.png)

2. There is no overall upgrade status to indicate the current status, for example, the image have already pending due to backend limit timeout, but for a user I can't figure out whether the upgrade is ongoing or not
  ![image](https://user-images.githubusercontent.com/29251855/157451294-95154f83-50a2-4368-8756-dda5aea32c25.png)

### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: `master-fd872c99-head` (03/09)

### Verify Steps
Zero downtime upgrade guide: https://github.com/harvester/harvester/pull/1945

1. Prepare a 3 node harvester cluster with v1.0.0 release
2. Enable virtual network 
3. Create a vlan network `vlan1` and id `1`
4. Upload the latest master-head iso file to a http server 
5. ssh to management node 
6. Create version.yaml file and add the following content 
  ```
  apiVersion: harvesterhci.io/v1beta1
  kind: Version
  metadata:
    name: 1.0.2
    namespace: harvester-system
  spec:
    isoChecksum: xxx                              <-- the build ISO sha512 checksum
    isoURL: http://10.10.0.1/harvester/harvester.iso      <-- the build ISO
    minUpgradableVersion: 0.3.0
    releaseDate: "20211231"
    tags:
    - dev                 <--- `dev` tag is important
    - test
  ``` 
7. Refresh harvester dashboard page 
8. Click the Upgrade button 
9. Select the version and start the upgrade 
10. Refresh the page and check the upgrade status besides the namespace
![image](https://user-images.githubusercontent.com/29251855/157455597-b63f1961-2e32-4057-bea7-aa2e53b1bf2f.png)

**Comment by bk201** [comment]
> There is no overall upgrade status to indicate the current status, for example, the image have already pending due to backend limit timeout, but for a user I can't figure out whether the upgrade is ongoing or not

@WuJun2016 We can use condition `Completed` field to indicate overall status:
- `Unknown`: on-going.
- `False`: Fail
- `True`: Success

@TachunLin The rancher version is shorter in formal releases, I guess it's OK to keep what it is.

**Comment by WuJun2016** [comment]
<img width="662" alt="image" src="https://user-images.githubusercontent.com/24985926/158117101-2c6374e6-0e44-4a68-af69-9b5df581014d.png">

**Comment by WuJun2016** [comment]
<img width="1671" alt="image" src="https://user-images.githubusercontent.com/24985926/158157812-e749615b-53aa-4d94-a2d4-b62a131c0b27.png">

**Comment by TachunLin** [comment]
Verified **`fixed`** on `v1.0.1-rc2`. Close this issue


### Result
* Now we have the overall status at the top of the upgrade dashboard
* The dashboard layout can contain all information
![image](https://user-images.githubusercontent.com/29251855/161874577-840ccad5-1114-4b1d-bf71-06b42fc03944.png)

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: `v1.0.1-rc2`

### Verify Steps
https://github.com/harvester/docs/blob/42598035cfe79856f7e159d4c47c4d7d29178900/docs/upgrade/automatic.md

**Comment by TachunLin** [comment]
### Additional Context 

**Full Upgrade Dashboard UI Screenshots**

![image](https://user-images.githubusercontent.com/29251855/161979845-263c5b93-3cb2-40ef-afd2-0a60a5512ca7.png)

![image](https://user-images.githubusercontent.com/29251855/161979920-9962d0c7-47c6-4df0-9392-0eb7a5aaa524.png)

![image](https://user-images.githubusercontent.com/29251855/161981317-d1b51461-cb63-464b-9f75-0617a2452388.png)

![image](https://user-images.githubusercontent.com/29251855/161981804-c6118974-7ea1-4390-8e6e-146f24da379f.png)

![image](https://user-images.githubusercontent.com/29251855/161982103-8664f0d8-aa37-4d17-9fbc-fc4f64400b86.png)

![image](https://user-images.githubusercontent.com/29251855/161983835-c5924995-4a90-49b0-9b1b-3224f046f187.png)

![image](https://user-images.githubusercontent.com/29251855/161989198-f7e4c1ac-ad93-4bb3-a950-b5fe0dbfa93e.png)

![image](https://user-images.githubusercontent.com/29251855/161991095-220a3ff4-88bb-4db8-bf28-f1d3cdd23dff.png)
